### ISSUE TITLE
Georeference Bot 

### ISSUE THREAD

**Issue by jldunnum** [ArctosDB/arctos#6151]
Georeference bot - please destroy this thing and put things back the way they were before it came along and added tons of really poor data to Arctos! 
Long list of collections upset about this. Another example attached. Not only did it georeference this locality totally wrong but then because its incorrect georeference put the locality in a different province it tells me I should change my higher geography to match. Unbelievable!

I also can't tell which of my own georeferences have been over written with incorrect data, and I shouldn't have to. Each collection has their work flow to go through and assess/update this stuff. This bot has thrown that to the wind. I simply do not have the bandwidth to vet all these and every time I find another misplaced locality my confidence in everything I have in there goes down because I don't know what has been changed. This is untenable, and I don't have time to go through months of discussions on the merits of automation, pure and simple it is hurting the quality of my collection data and increasing my data cleaning workload..
I'm serious, I need all MSB mammal localities georeferenced by the bot to revert to whatever value was there prior (even if that was a null value). 

![image](https://user-images.githubusercontent.com/17278851/232115408-d514276d-5b1d-4573-9c4e-690dfb15a14e.png)

Priority: High

**Comment by ebraker** [comment]
I'm with Jon here and requesting this for UCM (not sure if this rollback is Arctos-wide or per collection request... from #4916 it seems many people are in a similar boat). 

I do think there is potential for using a bot if we add additional fields and steps, e.g. implementing a new set of "suggested coordinate" fields available in the edit/curatorial view only; then once verified, an operator can click some sort of button to push vetted coordinates into the locality coordinate fields, OR if incorrect, click some sort of 'reviewed' box and annotate why suggested coordinates are not a good fit.

**Comment by dustymc** [comment]
It's by collection - I don't think this necessitates making decisions for users who don't exist yet....

>of "suggested coordinate" fields available in the edit/curatorial view only;

https://github.com/ArctosDB/arctos/issues/4916#issuecomment-1502181316

> 'reviewed' box a

There's a determiner and the bot cannot touch verificationstatus.

**Comment by ebraker** [comment]
Ok, then please remove for all UCM collections. 

> It's by collection - I don't think this necessitates making decisions for users who don't exist yet....

I think this needs to be discussed at the next meeting or even sooner. This bot is deeply distressing in terms of data quality and I very seriously doubt anyone will want this functionality as it currently stands, especially once viewing a random sample of their records, most of which will be wildly incorrect. I am extremely concerned and I think publishing the bot coordinate data how they are currently packaged pushes up against data ethics and threatens the validity of Arctos. This is so far from research grade and it is so difficult for a user to discern where this information is coming from.  Verification is not an optional part of the georeferencing process. Sorry but I have lost a lot of sleep over this and question if I should currently pull my data offline until this is resolved.

> There's a determiner and the bot cannot touch verificationstatus.

Agreed, this would require building functionality that does not currently exist.

**Comment by dustymc** [comment]
OK, I will stop work on this; I don't want to build something that may not have a reason to exist before it's built.

**Comment by dustymc** [comment]
I'm going back active on this for a few reasons:

1. This is obviously an emotional issue for some, I don't think we're going to get to any big-picture discussions very soon, this isn't quite an "easy fix" but it seems a price worth paying for the benefit
2. It's all optional, just a shortcut for something that can already be done, that nobody has to use, I don't think those can be controversial (other than regarding resources and etc, but see above, I think this is of sufficient priority)
3. I remain hopeful that The Community will decline to deny a tool to current and future users; in my view that should just never be on the table, and anything that might help us not attempt to cross that line seems worth prioritizing.

**Comment by DerekSikes** [comment]
Do the people having problems with this know that it's optional? I have not been paying attention to this issue because I never enabled this bot... but this makes me wonder about shared localities. What happens when a locality is shared by 2 collections and one of them has enabled this bot and the other hasn't? Couldn't it corrupt the data of both or does it know to only auto-georeference localities that are not shared outside bot-enabled collections?

**Comment by ebraker** [comment]
> The de-bot-tool would be optional.

I strongly think the opposite should be true - this should be an opt-in feature. Perhaps for shared localities (which generally indicate low complexity), the bot may do a decent job - though I have my doubts about this but can concede this point. However, I think collections should opt IN to using the bot for their data due to the aforementioned concerns, which are far from trivial and could be a major issue Arctos. I have opted out for UCM, but I am still extremely concerned for all Arctos data as it currently stands and the downstream impacts of publishing unverified coordinate data when "automated georeference" is not recorded in the suite of coordinate fields, can be easily excluded from downloads, and doesn't have a huge generic caveat remark that underscores the data are not verified.

There is no harm in opting in. This allows informed sanctioning by collections. The opposite is true at the moment.

**Comment by DerekSikes** [comment]
Ok, how do I find any localities in the data I manage that might have been auto-georeferenced so I can check them?

**Comment by dustymc** [comment]
>opt-in feature

Yea, probably. 

>find any localities

https://github.com/ArctosDB/arctos/issues/4916#issuecomment-1278163432

**Comment by ebraker** [comment]
@DerekSikes all of your localities have been georeferenced that did not have coordinate data to begin with, but you can also query:
![image](https://user-images.githubusercontent.com/11336485/233688616-4f6e9af0-3be6-42fd-9d69-b9ebb031d9c3.png)

**Comment by campmlc** [comment]
I'm sure that they do not know it is optional, or they would have disabled
it already. And that is an important point about shared localities.
This should be kept off until not just the technical issues but the
community issues are resolved. Yes, everyone is emotional - because they
aren't being listened to and they have no control over what's happened to
their data. So let's pause this.

**Comment by DerekSikes** [comment]
Yikes, ~2k of my catalog records and about 500 localities in UAM:Ento alone that I need to double-check? If only I had a collections manager I could assign this work to. Seems like dozens of hours of work.

**Comment by jldunnum** [comment]
Thanks Dusty, now on me to get everything georeferenced well on our own.

______________________________________________________________

Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by amgunderson** [comment]
Both Link and I also hate this and want it to stop and undo everything done to UAM:Mamm localities. I objected before and again and now an seeing bad coordinates. This never should have happened and I don't understand why it did.

**Comment by linkolson** [comment]
Yep. Too many unscrupulous "scientists" out there too lazy to confirm erroneous outliers or apply a simple biological smell test, leaving us to point out errors and ask for errata. This actually happened to us (UAM Mammals) in the past month. #killthebot

**Comment by AJLinn** [comment]
Please revert all UAM:EH localities as well. I've got unrequested stacks of information in specific localities and coordinates that seem to be all over the place in the random sampling I just checked.

**Comment by dustymc** [comment]
> objected before...don't understand

Clearly there's some gap in communication. https://github.com/ArctosDB/internal/issues/258 is intended to address that, input is most welcome.

UAM:Mamm and UAM:EH added to the list.

**Comment by DerekSikes** [comment]
I agree!

**Comment by jldunnum** [comment]
Also agree, and then we put out a notice letting collections know of the option to then reimplement where they want
 to.
______________________________________________________________

Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by dustymc** [comment]
Well that's depressing on a couple fronts, but it is a simplification so OK.

**Comment by dustymc** [comment]
Why?

**Comment by campmlc** [comment]
Just to announce to people that the bot is being discontinued, but they can still see what it did to their georeferences in test.
But I agree with undoing it and making it an opt in. Would it be possible to use it optionally for a designated set of specimens- -e.g for a single accession using named localities? My understanding is that right now we would have to georeference every locality individually, and being able to have the bot provide suggested coordinates for a group of localities, which can be optionally accepted, would be useful.

**Comment by dustymc** [comment]
We should probably go to Zoom if more discussion is needed, this whole situation feels like a near-perfect failure to communicate.

I do not have available any development pathway that doesn't begin with test.

The data that the bot uses have been visible and available for use for many years.

The bot does nothing to named localities.

The bot does nothing to existing spatial data, wherever it might have come from.

I am very hesitant to increase the complexity of any automation in any way at this point, but it would be trivial to script specified localities on request.

**Comment by amgunderson** [comment]
The georeference bot claims this wolf was swimming in the Pacific Ocean...bad coordinates are worse than no coordinates.
<img width="1338" alt="Screenshot 2023-04-25 at 11 49 16 AM" src="https://user-images.githubusercontent.com/8440698/234387434-6ae467b3-bb83-44ee-a572-19388911b536.png">

**Comment by ebraker** [comment]
@dustymc just saw this in one of my records:
![image](https://github.com/ArctosDB/arctos/assets/11336485/d86b107b-2239-4af0-845e-860a2333c732)

From our conversation above I thought this was going to be an opt-in feature moving forward (or paused altogether until we figure out a better approach), so a georeference source seems unnecessary since 'do not automatically georeference' would be the default for most users(?). If not, can we please mask? This note will be on nearly all of my legacy records (tens of thousands) and I'd rather it not be visible.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
@dustymc one of our localities (locality name NMMNH:Paleo:L-12916) had coordinates deleted by unknown on April 28th, the same day this issue was closed. When I created the locality in November the bot added coordinates, which I then immediately edited because I actually had coordinates. Could this locality have been accidently caught up in removing bot coordinates?

I've been trying to figure out if I could have done something that could have deleted the coordinates, but I think I've only ever done the one bulk edit back in July last year.
